Tempo: Update docs for trace to logs functionality (#62338)
grafana/grafana: docs/sources/datasources/tempo/_index.md

The **Trace to logs** setting configures the [trace to logs feature]({{< relref "../../explore/trace-integration" >}}) that is available when you integrate Grafana with Tempo.

> There are two ways to configure the trace to logs feature. You can use simplified configuration with default query, or you can configure custom query where you can use a template language to interpolate variables from the trace or span.

Reviewer: Might not be totally clear what template language means in this case

Author: Added a link to the variable docs

Reviewer: Looks good

**To use simple configuration:**

1. Select the target data source.
1. Set start and end time shift. As the logs timestamps may not exactly match the timestamps of the spans in trace it may be necessary to search in larger or shifted time range to find the desired logs.
1. Select which tags to use in the logs query. The tags you configure must be present in the spans attributes or resources for a trace to logs span link to appear. You can optionally configure a new name for the tag. This is useful for example if the tag has dots in the name and the target data source does not allow using dots in labels. In that case you can for example remap `http.status` to `http_status`.
1. Optionally switch on Filter by Trace ID and/or Filter by Span ID to further filter the logs if your logs consistently contain trace or span IDs.

**To use custom query configuration:**

1. Select the target data source.

> 1. Set start and end time shift. As the logs timestamps may not exactly match the timestamps of the spans in trace it may be necessary to search in larger or shifted time range to find the desired logs.

Reviewer 2: ```suggestion
1. Set start and end time shift. As the logs timestamps may not exactly match the timestamps of the spans in the trace it may be necessary to widen or shift the time range to find the desired logs.
```

> 1. Optionally select tags to map. In the custom query you can use all tags separately, even those not mapped here or you can use ${\_\_tags} variable which will use and interpolate the tags mapped here. The tags you configure must be present in the spans attributes or resources for a trace to logs span link to appear. You can optionally configure a new name for the tag. This is useful for example if the tag has dots in the name and the target data source does not allow using dots in labels. In that case you can for example remap `http.status` to `http_status`.

Reviewer: `even those not mapped here` is a bit unclear

Author: I reworded the description here, please let me know if it is clearer like this.

Reviewer: Thanks. This is much clearer.

> 1. Optionally select tags to map. In the custom query you can use all tags separately, even those not mapped here or you can use ${\_\_tags} variable which will use and interpolate the tags mapped here. The tags you configure must be present in the spans attributes or resources for a trace to logs span link to appear. You can optionally configure a new name for the tag. This is useful for example if the tag has dots in the name and the target data source does not allow using dots in labels. In that case you can for example remap `http.status` to `http_status`.

Reviewer 2: ```suggestion
1. Optionally select tags to map. In the custom query you can use all tags separately, even those not mapped here or you can use ${\_\_tags} variable which will use and interpolate the tags mapped here. The tags you configure must be present in the span's attributes or resources for a trace to logs span link to appear. You can optionally configure a new name for the tag. This is useful in cases where the tag has dots in the name and the target data source does not allow using dots in labels. For example, you can remap `http.status` to `http_status` in such a case.
```

| \_\_span.tags      | Namespace for the tags in the span. To access a specific tag named "version" you would use `${__span.tags.version}`. In case the tag contains dot you have to access it as `${__span.tags["http.status"]}`                                                                                                                                                         |
| \_\_trace.traceId  | The ID of the trace.                                                                                                                                                                                                                                                                                                                                               |
| \_\_trace.duration | The duration of the trace                                                                                                                                                                                                                                                                                                                                          |

> | \_\_trace.name     | The name of the trace.                                                                                                                                                                                                                                                                                                                                             |

Reviewer: Only some of these descriptions have a full stop
